src/hotspot/share/opto/phasetype.hpp

```diff
@@ -168,6 +168,8 @@ class PhaseNameValidator {
         // strncpy always writes len characters. If the source string is shorter, the function fills the remaining bytes with NULLs.
         strncpy(_bad, *iter, len);
         _valid = false;
+      } else if (PHASE_DEBUG == cpt) {
```

Comment:
I don't think abusing the unused `PHASE_DEBUG` in this way is a good and clean approach. I'd rather add a new enum entry `PHASE_ALL` or something like that in `CompilerPhaseType` as you have proposed.